File: 0739-daily-temperatures/0739-daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        int n = temperatures.size();
        vector<int> res(n, 0);
        stack<int> mSt;
        for (int i=n-1; i>=0; --i) {
            while (!mSt.empty() && temperatures[i] >= temperatures[mSt.top()])
                mSt.pop();
            res[i] = mSt.empty() ? 0 : (mSt.top() - i);
            mSt.push(i);
        }
        return res;
    }
};

File: 0739-daily-temperatures/0739-daily-temperatures.cs
public class Solution {
    public int[] DailyTemperatures(int[] temperatures) {
        int n = temperatures.Length;
        int[] res = new int[n];
        Stack<int> st = new Stack<int>();
        
        for (int i = n - 1; i >= 0; --i) {
            while (st.Count > 0 && temperatures[i] >= temperatures[st.Peek()]) {
                st.Pop();
            }
            res[i] = st.Count == 0 ? 0 : st.Peek() - i;
            st.Push(i);
        }
        
        return res;
    }
}
